src/System.Net.Primitives/src/System/Net/IPEndPoint.cs

```diff
@@ -119,6 +119,66 @@ public int Port
             }
         }
 
+        public static bool TryParse(string s, out IPEndPoint result)
+        {
+            return TryParse(s.AsSpan(), out result);
+        }
+
+        public static bool TryParse(ReadOnlySpan<char> s, out IPEndPoint result)
+        {   
+            int addressLength = s.Length;  // If there's no port then send the entire string to the address parser
+            int lastColonPos = s.LastIndexOf(':');
+
+            // Look to see if this is an IPv6 address with a port.
+            if (lastColonPos > 0)
+            {
+                if (s[lastColonPos - 1] == ']')
+                {
+                    addressLength = lastColonPos;
+                }
+                // Look to see if this is IPv4 with a port (IPv6 will have another colon)
+                else if (s.Slice(0, lastColonPos).LastIndexOf(':') == -1)
+                {
+                    addressLength = lastColonPos;
+                }
+            }
+
+            if (IPAddress.TryParse(s.Slice(0, addressLength), out IPAddress address))
+            {
+                int port = 0;
+                if (addressLength == s.Length ||
+                    (int.TryParse(s.Slice(addressLength + 1), out port) && port >= MinPort && port <= MaxPort))
```

Comment:
InvariantCulture?

Reply:
We should change this to use `uint` instead of `int` and pass `NumberStyles.None` as the style... at which point the culture should be ignored, and we're protected against things like an invalid `+` at the beginning of the number.

Reply:
(If we don't have one already, it'd also be good to add tests that validate ports with +/- at the beginning fail.)

Reply:
https://github.com/dotnet/corefx/blob/e73815dec37fa5df36c7e865889478a62fedb507/src/System.Net.Primitives/tests/FunctionalTests/IPEndPointParsing.cs#L126

Adding "+" test now...